lib/vault/client.rb

```diff
@@ -180,6 +180,19 @@ def request(verb, path, data = {}, headers = {})
       # crazy strange conditionals.
       connection = Net::HTTP::Persistent.new(name: "vault-ruby")
 
+      if proxy_address
+        proxy_uri = URI "http://#{proxy_address}"
```

Comment:
I know it's little, but can we use parens here or URI.parse?

I'm also worried that we are prefixing with http...this would make it impossible to use an HTTPS proxy address.